Core/OfficeDevPnP.Core/Utilities/WebParts/IWebPartPostProcessor.cs

```diff
@@ -5,6 +5,6 @@ namespace OfficeDevPnP.Core.Utilities.WebParts
 {
     public interface IWebPartPostProcessor
     {
-        void Process(WebPartDefinition wpDefinition, File webPartPage);
+        void Process(WebPartDefinition wpDefinition, Web web);
```

Comment:
Hi Paolo, 
Ok, no problems.   

Unfortunately `WebId` is not available for on-premise and current fix is the only possible I found. Let me give you different options and you will advise which is best suited for you: 
 1. Inside `IWebPartPostProcessor` create a new method with the same name, but different param. Mark original as deprecated or obsolete.  Drawbacks - every class needs to implement obsolete interface method (will have empty body or throw an exception).   

2. Create new interface and mark `IWebPartPostProcessor` obsolete. Drawbacks - return value for `WebPartPostProcessorFactory` will be changed to different interface.     
3. Do nothing and wait for next major update for the core library, where we can change the signature and say that since that's a major release, some methods are changed.
3. May be I missed some other options, if you can provide one, that would be perfect.   
  
Personally I love `#2` but you choose.